tembo-operator/src/controller.rs

```diff
@@ -378,23 +383,16 @@ impl CoreDB {
                     last_fully_reconciled_at: None,
                 }
             }
+            // If the cluster is stopped, set the status to a very sparse set
+            // of information that will allow other provisioning teps to
+            // complete, but otherwise do nothing else. The CNPG operator will
+            // handle the rest.
             true => {
-                let current_config_values = get_current_config_values(self, ctx.clone()).await?;
-                CoreDBStatus {
-                    running: false,
-                    extensionsUpdating: false,
-                    storage: Some(self.spec.storage.clone()),
-                    extensions: self.status.as_ref().and_then(|f| f.extensions.clone()),
-                    trunk_installs: self.status.as_ref().and_then(|f| f.trunk_installs.clone()),
-                    resources: Some(self.spec.resources.clone()),
-                    runtime_config: Some(current_config_values),
-                    first_recoverability_time: self
-                        .status
-                        .as_ref()
-                        .and_then(|f| f.first_recoverability_time),
-                    pg_postmaster_start_time: None,
-                    last_fully_reconciled_at: None,
-                }
+                let mut status = self.status.clone().unwrap_or_default();
```

Comment:
I would not use `unwrap_or_default` anywhere.  If for some reason if status is empty this will panic, or potentially panic.  A possibly solution might be this, but I would make sure this works before applying it.

```rust
let status = match self.status.clone() {
    Some(s) => s,
    None => return Err(Action::requeue(Duration::from_secs(300)))
};
```

Reply:
unwrap or default doesn't panic when unwrap fails, instead it returns the default of the desired type (CoreDBStatus)